### PR TITLE
MINOR: resolve warnings in KStreamImpl

### PR DESCRIPTION
**Pull request by vvcephei**
We shouldn't be ignoring compiler warnings, since they tell us when our code is likely incorrect.

Sadly, our build produces hundreds of warnings in Streams alone. I plan to just go down the list and resolve them every now and then.

This is the second installment: KStreamImpl.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by vvcephei** [comment]
@mjsax @guozhangwang ,

I had a slow day, so I thought I'd try to resolve the compiler warnings in the main `streams` project. I've chunked them up into 4 manageable changesets.

Do you mind reviewing this one?

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java [hunk 1]
```diff
@@ -161,40 +166,44 @@ public KStreamImpl(final InternalStreamsBuilder builder,
...
+        final String name = builder.newProcessorName(KEY_SELECT_NAME);
 
 
-        KStreamMap kStreamMap = new KStreamMap<>(
+        final KStreamMap<K, V, K1, V> kStreamMap = new KStreamMap<>(
             (KeyValueMapper<K, V, KeyValue<K1, V>>) (key, value) -> new KeyValue<>(mapper.apply(key, value), value));
 
 
-        ProcessorParameters<K1, V> processorParameters = new ProcessorParameters<>(kStreamMap, name);
+        final ProcessorParameters<K, V> processorParameters = new ProcessorParameters<>(kStreamMap, name);
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:183]: Nice catch.

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java [hunk 2]
```diff
@@ -356,37 +382,54 @@ public void print(final Printed<K, V> printed) {
...
     }
 
     @Override
     public void foreach(final ForeachAction<? super K, ? super V> action) {
         Objects.requireNonNull(action, "action can't be null");
-        String name = builder.newProcessorName(FOREACH_NAME);
+        final String name = builder.newProcessorName(FOREACH_NAME);
 
-        ProcessorParameters processorParameters = new ProcessorParameters<>(new KStreamPeek<>(action, false), name);
+        final ProcessorParameters processorParameters = new ProcessorParameters<>(
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:401]: nit: is it necessary for triggering a warning? My personal preference is that if a single line is not too long, then it is okay to do so.

Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:401]: ditto below.

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java [hunk 3]
```diff
@@ -584,32 +651,38 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
...
-                                                                   valSerde,
-                                                                   null,
-                                                                   name,
-                                                                   repartitionNodeBuilder);
-
-        OptimizableRepartitionNode<K, V> repartitionNode = repartitionNodeBuilder.build();
+        final OptimizableRepartitionNode.OptimizableRepartitionNodeBuilder<K, V> repartitionNodeBuilder =
+            OptimizableRepartitionNode.optimizableRepartitionNodeBuilder();
+
+        final String repartitionedSourceName = createRepartitionedSource(
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:676]: This is a meta question: Is this re-alignment with the first parameter in new line necessary for removing the warning?

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java [hunk 4]
```diff
@@ -752,17 +849,25 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
...
+            name,
+            new KStreamKTableJoin<>(((KTableImpl<K, ?, V1>) other).valueGetterSupplier(), joiner, leftJoin),
+            this.name
+        );
         builder.internalTopologyBuilder.connectProcessorAndStateStores(name, ((KTableImpl) other).valueGetterSupplier().storeNames());
 
-        ProcessorSupplier<K, V> processorSupplier = new KStreamKTableJoin<>(((KTableImpl<K, ?, V1>) other).valueGetterSupplier(),
-                                                                            joiner,
-                                                                            leftJoin);
+        final ProcessorSupplier<K, V> processorSupplier = new KStreamKTableJoin<>(
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:859]: This is not introduced by this PR but: `processorSupplier` can be reused for `addProcessor` and `ProcessorParameters` constructor below for both the physical and logical plan generation.

Similarly the storeNames can be reused for both as well.

**Review by guozhangwang — COMMENTED** [review]
Some minor comments and a meta question about parameter alignment.

**Comment by bbejeck** [comment]
cherry picked this to https://github.com/apache/kafka/pull/5201

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java (hunk 3 above)
Comment (vvcephei) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:676]: no, it's purely reformatting.

I thought that the code style specified that we should start new line for the first parameter, but it actually doesn't comment on it.

I can just adjust my code formatter to permit the prior style: allow leaving the first param on the same line as the method call and align the remaining parameters with the first.

**Comment by vvcephei** [comment]
Closing until after #5201 is merged.

**Pull request closed without merging**